flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1640,13 +1640,102 @@ struct InsertValueOpConversion
 
 /// InsertOnRange inserts a value into a sequence over a range of offsets.
 struct InsertOnRangeOpConversion
-    : public FIROpAndTypeConversion<fir::InsertOnRangeOp> {
+    : public FIROpAndTypeConversion<fir::InsertOnRangeOp>,
+      public ValueOpCommon {
   using FIROpAndTypeConversion::FIROpAndTypeConversion;
 
+  // Increments an array of subscripts in a row major fasion.
+  void incrementLoop(const SmallVector<uint64_t, 8> &dims,
```

Comment:
`incrementSubscripts` would be a clearer name in my opinion.

Reply:
I agree.